internal/database/database_test.go

```diff
@@ -271,8 +289,13 @@ func TestDatabase(t *testing.T) {
 					Name:         "system6",
 					Requirements: config.Requirements{},
 				}),
+				newTestCase("delete bundle and sources"). // We check that it deletes cross-refs by attempting to delete those before and after the bundle delete: TODO(sr)
+										DeleteSource("source-a", false). // Cannot delete source, it's referenced
```

Comment:
> One thing that might be nice to have is a test of the "can't delete a source if it's referenced in a bundle or a stack" behavior.

This is it. Deleting "source-a" fails at first because it's references in "bundle-a". After that's deleted, we'll be able to successfully delete "source-a".


I'll add a case for stack -> source, good point.